### ISSUE TITLE
Dropdown field doesn't show as empty when using the dropdownBuilder

### ISSUE THREAD

**Issue by GP4cK** [salim-lachdhaf/searchable_dropdown#515]
**Describe the bug**
I was following your [instructions](https://github.com/salim-lachdhaf/searchable_dropdown/issues/500) to have the search text field inside the dropdown field by using the `dropdownBuilder`.
However, the dropdown field never shows the empty state because of this line: https://github.com/salim-lachdhaf/searchable_dropdown/blob/9f1b1d67b62a2a6f23caf520c4fb078f02027312/lib/dropdown_search.dart#L394

Would it be ok to change it to?

```diff
- isEmpty: getSelectedItem == null && widget.dropdownBuilder == null,
+ isEmpty: getSelectedItem == null
```

I can make a PR if it's fine with you.

**To Reproduce**
1. Create a dropdown field like this:
```dart 
DropdownSearch(
  dropdownDecoratorProps: DropDownDecoratorProps(
    dropdownSearchDecoration: InputDecoration(labelText: 'Hello'),
  ),
  dropdownBuilder: (context, selectedItem) => const TextField(decoration: null),
);
```
2. See that the field doesn't show as empty

**Expected behavior**
The field should show as empty with the label in the center of the box.

**Screenshots**
![image](https://user-images.githubusercontent.com/2680243/204725636-df1160ee-7001-42bf-8e1c-58bffc4a73db.png)

**Comment by salim-lachdhaf** [comment]
i will see what i can do to achieve that need.

**Comment by salim-lachdhaf** [comment]
we MUST not force user to not see null value if he he wants see it.

the solution in this case is to developper to decide, for example if the user don't want see null , simply he use this code:
```

DropdownSearch(
                items: ["Test1", "Test2"],
                dropdownDecoratorProps: DropDownDecoratorProps(
                  dropdownSearchDecoration: InputDecoration(
                    labelText: "Enter Name",
                    labelStyle: TextStyle(color: Colors.black45),
                    border: OutlineInputBorder(
                      borderSide: const BorderSide(
                          color: Colors.deepPurple, width: 1.0),
                      borderRadius: BorderRadius.circular(8),
                    ),
                  ),
                ),
                dropdownBuilder: (context, selectedItem) {
                  return selectedItem==null ? SizedBox.shrink() : Text(selectedItem.toString()); //here it's to developer to decide what the lib should shows
                },
              ),
```

**Comment by GP4cK** [comment]
The solution provided doesn't work because you're still passing a `dropdownBuilder`.
This means that `isEmpty` of the InputDecorator will always be false and the label will always be shown on top instead of inside the field.

https://github.com/salim-lachdhaf/searchable_dropdown/blob/1aa46cb49115c33eacd562ce0fe1931e6ac98d15/lib/dropdown_search.dart#L401-L402

So the workaround I found was to use a ValueNotifier to track if the dropdown is empty and rebuild in case it changes.
It looks like this:

```dart
```

**Comment by GP4cK** [comment]
I feel like the `isEmpty` could just be `getSelectedItem == null && getSelectedItems.isEmpty`.
Then if people want the label to stay on top, they can just do:

```dart
DropdownSearch(
  decorationProps: DropDownDecorationProps(
    decoration: InputDecoration(
      floatingLabelBehavior: FloatingLabelBehavior.always,
    ),
  ),
),
```

**Comment by salim-lachdhaf** [comment]
@GP4cK Good point, i think either `floatingLabelBehavior` should be the best choice of user for whatever keep the label as hint or always on top ...
It will be fixed in 6.0.1 that will be released in next days.

**Comment by salim-lachdhaf** [comment]
fixed in 6.0.1, it will be released today.
